src/server/chromium/crNetworkManager.ts

```diff
@@ -165,6 +165,22 @@ export class CRNetworkManager {
       });
     }
     if (!event.networkId) {
+      // Some downloads come in as document requests without a network id.
+      // Forge a requestWillBeSent event for them so that users can handle them.
+      if (event.resourceType === 'Document') {
```

Comment:
I looked at this issue. This direction seems fine, but we have to go further. For example, if we allow intercepting this request, we must ensure that it gets the corresponding request/response/requestfinished/requestfailed events with all the data inside. Basically, let's forge all the events when `networkId` is missing. Note that in the case of interception, we have to forge events after the interceptor runs.